lib/async/http/faraday/adapter.rb

```diff
@@ -181,6 +181,12 @@ def perform_request(env)
 						end
 						
 						if headers = env.request_headers
+              # Ignore Content-Length if given, it will be set for us later anyway (lowercased)
```

Comment:
This is looking good.

I'd suggest extracting this into `length` and then assigning to `BodyReadWrapper`, which should have this:

```
class BodyReadWrapper
  def initialize(body, length = nil, block_size: 4096)
    @body = body
    @length = length
    @block_size = block_size
  end

  # ...

  def length
    @length # if known
  end
```

You'll need to move the `if headers = env.request_headers` first to extract a local variable if possible.

Is `headers.key?(...)` case sensitive?

Testing is a bit tricky, but maybe we can mock the client. If you can get the code updated, I can help write the tests.

Reply:
@ioquatix 

> Is headers.key?(...) case sensitive?

Yes. I'd change this to check for both `Content-Length` and `content-length` if the intent is to use that value for `BodyReadWrapper#length`, if you agree.